File: src/hwacc/ComputeUnit.py
from m5.params import *
from m5.proxy import *
from m5.SimObject import SimObject
from m5.objects.CommInterface import CommInterface
from m5.objects.HWInterface import HWInterface

class ComputeUnit(SimObject):
    type = 'ComputeUnit'
    cxx_header = "hwacc/compute_unit.hh"

    comm_int = Param.CommInterface(Parent.any, "Communication interface to connect to")
    hw_int = Param.HWInterface(Parent.any, "Hardware model interface to connect to")
    

File: src/hwacc/compute_unit.cc
//------------------------------------------//
#include "hwacc/compute_unit.hh"
//------------------------------------------//

ComputeUnit::ComputeUnit(const ComputeUnitParams &p) :
    SimObject(p),
    comm(p.comm_int),
    hw(p.hw_int),
    tickEvent(this) {}

// ComputeUnit*
// ComputeUnitParams::create() {
//     return new ComputeUnit(this);
// }
